servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -326,6 +362,37 @@ public StreamingHttpClient buildStreaming() {
         }
     }
 
+    static StreamingHttpRequestResponseFactory defaultReqRespFactory(ReadOnlyHttpClientConfig roConfig,
```

Comment:
Can we reuse this method inside `buildStreaming` for `reqRespFactory ` instead of duplicating the logic?

Reply:
good point, some duplication of conditionals but shouldn't be a big issue